### PR TITLE
docs: translate "State: A Component's Memory"

### LINKED ISSUE #446: State: A Component's Memory
Page: https://id.react.dev/learn/state-a-components-memory

Assignee: @krowter

### PR DESCRIPTION
**Pull request by krowter**
Closes #446
Page link https://react.dev/learn/state-a-components-memory

### CODE REVIEW

File: src/content/learn/state-a-components-memory.md [hunk 1]
```diff
@@ -4,22 +4,22 @@ title: "State: A Component's Memory"
 
 <Intro>
 
-Components often need to change what's on the screen as a result of an interaction. Typing into the form should update the input field, clicking "next" on an image carousel should change which image is displayed, clicking "buy" should put a product in the shopping cart. Components need to "remember" things: the current input value, the current image, the shopping cart. In React, this kind of component-specific memory is called *state*.
+Seringkali komponen mengubah tampilan di layar sebagai respons terjadinya interaksi dari pengguna. Mengetik di dalam form akan memperbarui kolom masukan, menekan tombol "Selanjutnya" pada slide gambar akan mengganti gambar yang ditampilkan, menekan "Beli" akan menambahkan barang ke dalam keranjang. Komponen perlu "mengingat" informasi-informasi ini: nilai kolom input, gambar, dan isi keranjang belanja. Dalam React, ingatan yang dimiliki komponen ini disebut *state*.
```
Comment (mhaidarhanif) [inline src/content/learn/state-a-components-memory.md:7]: To adhere with the source with `need`.

Let's also introduce the original English word `input field` once in the beginning.

```suggestion
Seringkali komponen perlu mengubah tampilan di layar sebagai respons terjadinya interaksi dari pengguna. Mengetik di dalam form akan memperbarui kolom masukan (*input field*), menekan tombol "Selanjutnya" pada slide gambar akan mengganti gambar yang ditampilkan, menekan "Beli" akan menambahkan barang ke dalam keranjang. Komponen perlu "mengingat" informasi-informasi ini: nilai kolom input, gambar, dan isi keranjang belanja. Dalam React, ingatan yang dimiliki komponen ini disebut *state*.
```

**Review by mhaidarhanif — CHANGES_REQUESTED** [review]
An improvement.

**Comment by krowter** [comment]
Just got the chance to work on this again 😁
Suggestions committed.

File: src/content/learn/state-a-components-memory.md [hunk 2]
```diff
@@ -4,22 +4,22 @@ title: "State: A Component's Memory"
...
 
 </Intro>
 
 <YouWillLearn>
 
-* How to add a state variable with the [`useState`](/reference/react/useState) Hook
-* What pair of values the `useState` Hook returns
-* How to add more than one state variable
-* Why state is called local
+* Bagaimana menambahkan _state_ dengan Hook [`useState`](/reference/react/useState)
```
Comment (r17x) [inline src/content/learn/state-a-components-memory.md:13]: Gunakan "*" dari pada "_", Related: https://github.com/reactjs/id.react.dev/pull/531#discussion_r1186808235

File: src/content/learn/state-a-components-memory.md [hunk 3]
```diff
@@ -151,54 +152,55 @@ button {
 
 </Sandpack>
 
-The `handleClick` event handler is updating a local variable, `index`. But two things prevent that change from being visible:
+_event handler_ `handleClick` memperbarui nilai variabel `index`. Namun dua hal mencegah pembaruan tersebut ditampilkan ke pengguna:
 
-1. **Local variables don't persist between renders.** When React renders this component a second time, it renders it from scratch—it doesn't consider any changes to the local variables.
-2. **Changes to local variables won't trigger renders.** React doesn't realize it needs to render the component again with the new data.
+1. **Variabel lokal tidak dipertahankan antarrender.** Saat React merender komponen ini untuk kedua kalinya, dia membuatnya ulang dari awal—tidak memerhatikan adanya perubahan ke variabel tersebut.
+2. **Perubahan terhadap variabel lokal tidak memicu render.** React tidak menyadari kalau dia perlu melakukan render ulang dengan data yang baru.
```
Comment (r17x) [inline src/content/learn/state-a-components-memory.md:158]:
```suggestion
2. **Perubahan terhadap variabel lokal tidak memicu *render*.** React tidak menyadari kalau dia perlu melakukan *render* ulang dengan data yang baru.
```

File: src/content/learn/state-a-components-memory.md [hunk 4]
```diff
@@ -891,33 +895,33 @@ button {
...
+Perhatikan juga bagaimana komponen `Page` tidak "mengetahui" tentang state milik `Gallery` atau bahkan ada-tidaknya. Tidak seperti props, **state bersifat privat ke komponen tempat dia dideklarasikan.** Komponen *parent* tidak dapat mengubahnya. Sehingga Anda bisa menambahkan atau menghapus state tanpa memengaruhi komponen lainnya.
 
-What if you wanted both galleries to keep their states in sync? The right way to do it in React is to *remove* state from child components and add it to their closest shared parent. The next few pages will focus on organizing state of a single component, but we will return to this topic in [Sharing State Between Components.](/learn/sharing-state-between-components)
+Bagaimana jika Anda ingin menjaga state di kedua `Gallery` tetap sinkron? Cara yang benar dalam React adalah *menghapus* state dari komponen *child* dan memindahkannya ke komponen *parent* terdekat yang sama. Beberapa halaman berikutnya akan fokus ke mengatur state dalam sebuah komponen, tapi kita akan kembali ke topic ini di [Sharing State Between Components.](/learn/sharing-state-between-components)
```
Comment (r17x) [inline src/content/learn/state-a-components-memory.md:902]:
```suggestion
Bagaimana jika Anda ingin menjaga state di kedua `Gallery` tetap sinkron? Cara yang benar dalam React adalah *menghapus* *state* dari komponen anak  (*child*) dan memindahkannya ke komponen Induk (*parent*) terdekat yang sama. Beberapa halaman berikutnya akan fokus ke mengatur *state* dalam sebuah komponen, tapi kita akan kembali ke topic ini di [Sharing State Between Components.](/learn/sharing-state-between-components)
```

**Review by r17x — CHANGES_REQUESTED** [review]
Permintaan perubahan meliputi:

* Gunakan `*` daripada `_` untuk membuat tulisan miring *italic*
* Buat tulisan miring pada kata serapan (bukan Bahasa Indonesia), seperti *render*, *state*, atau *props*.

File: src/content/learn/state-a-components-memory.md (hunk 1 above)
Comment (krowter) [reply src/content/learn/state-a-components-memory.md]: Updated. Thanks

File: src/content/learn/state-a-components-memory.md (hunk 2 above)
Comment (krowter) [reply src/content/learn/state-a-components-memory.md]: Updated for all italics ya

File: src/content/learn/state-a-components-memory.md (hunk 4 above)
Comment (krowter) [reply src/content/learn/state-a-components-memory.md]: Added "anak" and "Induk" before their english words

File: src/content/learn/state-a-components-memory.md [hunk 5]
```diff
@@ -545,18 +548,18 @@ function useState(initialState) {
   let pair = componentHooks[currentHookIndex];
   if (pair) {
     // This is not the first render,
-    // so the state pair already exists.
+    // so the *state* pair already exists.
     // Return it and prepare for next Hook call.
     currentHookIndex++;
```
Comment (resir014) [inline src/content/learn/state-a-components-memory.md:553]: Kalau ini di-italicise juga, kalau begitu comment di code example-nya sekalian ditranslate juga ya. @krowter

File: src/content/learn/state-a-components-memory.md [hunk 6]
```diff
@@ -1421,42 +1425,42 @@ export default function FeedbackForm() {
...
-Try moving the second `useState` call after the `if` condition and notice how this breaks it again.
+Coba pindahkan pemanggilan `useState` kedua ke bawah kondisi `if` dan perhatikan kodenya berhenti bekerja lagi. 
 
-If your linter is [configured for React](/learn/editor-setup#linting), you should see a lint error when you make a mistake like this. If you don't see an error when you try the faulty code locally, you need to set up linting for your project. 
+Jika *linter* Anda [disetel untuk React](/learn/editor-setup#linting), Anda seharusnya melihat pesan galat saat melakukan kesalahan seperti ini. Jika Anda tidak melihatnya, Anda perlu memasang *linter* untuk proyek Anda.
 
 </Solution>
 
-#### Remove unnecessary state {/*remove-unnecessary-state*/}
+#### Menghapus *state* yang tidak perlu {/*remove-unnecessary-state*/}
```
Comment (resir014) [inline src/content/learn/state-a-components-memory.md:1457]:
```suggestion
#### Menghapus state yang tidak perlu {/*remove-unnecessary-state*/}
```

Don't italicise loanwords in headings.

File: src/content/learn/state-a-components-memory.md [hunk 7]
```diff
@@ -724,15 +728,15 @@ button { display: block; margin-bottom: 10px; }
 
 </Sandpack>
 
-You don't have to understand it to use React, but you might find this a helpful mental model.
+Anda tidak perlu mendalaminya untuk menggunakan React, tapi bisa memberi Anda gambaran kasar tentang cara kerjanya.
 
 </DeepDive>
 
-## State is isolated and private {/*state-is-isolated-and-private*/}
+## *State* terisolasi dan privat {/*state-is-isolated-and-private*/}
```
Comment (resir014) [inline src/content/learn/state-a-components-memory.md:735]:
```suggestion
## State terisolasi dan privat {/*state-is-isolated-and-private*/}
```

Idem.

File: src/content/learn/state-a-components-memory.md [hunk 8]
```diff
@@ -520,19 +523,19 @@ button {
...
+Baiknya memang ada beberapa variabel *state* jika mereka tidak saling berhubungan, misal `index` dan `showMore` dalam contoh tadi. Tapi kalau Anda merasa dua *state* akan sering berganti nilai bersama, ada baiknya untuk menggabungkannya. Misal, jika Anda mempunyai form dengan beberapa kolom, akan lebih mudah jika ada satu variabel *state* berupa objek daripada ada variabel *state* untuk masing-masing kolom. Baca [Memilih struktur state](/learn/choosing-the-state-structure) untuk tips lainnya.
 
 <DeepDive>
 
-#### How does React know which state to return? {/*how-does-react-know-which-state-to-return*/}
+#### Bagaimana React tahu *state* mana yang harus dikembalikan? {/*how-does-react-know-which-state-to-return*/}
```
Comment (resir014) [inline src/content/learn/state-a-components-memory.md:530]:
```suggestion
#### Bagaimana React tahu state mana yang harus dikembalikan? {/*how-does-react-know-which-state-to-return*/}
```

Idem.

**Review by resir014 — CHANGES_REQUESTED** [review]
Some requested changes. @krowter 

Also, please change the page title to "State: Memori dari Sebuah Komponen"

**Comment by mhaidarhanif** [comment]
@krowter Please fix and resolve this to be approved bby @resir014 @r17x and me

**Review by mhaidarhanif — APPROVED** [review]

**Comment by mhaidarhanif** [comment]
I approved to avoid future blocking, now can be approved by @resir014 @r17x once the issues have been resolved

**Comment by krowter** [comment]
Sorry. I missed the notification email.
Will fix tomorrow morning, asap

File: src/content/learn/state-a-components-memory.md (hunk 5 above)
Comment (krowter) [reply src/content/learn/state-a-components-memory.md:553]: siap mas :+1:

**Comment by mhaidarhanif** [comment]
@resir014 @r17x can check and approve if everything's okay

**Review by resir014 — APPROVED** [review]

**Pull request merged**